enumerable.rb

```diff
@@ -0,0 +1,97 @@
+module Enumerable
+    def my_each
+      return to_enum unless block_given?
+
+        for item in self
+            yield(item)
+        end
+        self
+    end
+   
+    def my_each_with_index
+      return to_enum unless block_given?
+      
+        i = 0
+        while i < self.length
+         yield(self[i], i)
+        i += 1
+    end
+    self
+    end
+
+    def my_select
+      return to_enum unless block_given?
+
+        new_array = []
+        self.my_each do |item|
+          new_array.push(item) if yield(item)
+        end
+        new_array
+    end
+
+    def my_all?
+      return to_enum unless block_given?
+
+        test = true
+        self.my_each do |item|
+          test = false if !yield(item)
+          break
+        end
+        return test
+    end
+    
+    def my_any?
+      return to_enum unless block_given?
+
+        test = false
+        self.my_each do |item|
+          test = true if yield(item)
+          break
+        end
+        return test
+    end
+    
+    def my_none?
+      return to_enum unless block_given?
+      
+        test = true
+        self.my_each do |item|
+          test = false if yield(item)
+          break
+        end
+        return test
+    end
+    
+    def my_count
+        total = 0
+        if block_given?
+          self.my_each {|item| total += 1 if yield(item)}
+        else
+          total = self.count
+        end
+        return total
+    end
+    
+    def my_map(&my_proc)
+        new_array = []
+        self.my_each do |value|
+          if my_proc.nil?
+            new_array.push(my_proc.call(value))
+          else
+            new_array.push(yield(value))
+          end
+        end
+        return new_array
+    end
+    
+    def my_inject(memo=self[0])
+        self.my_each_with_index do |value, index|
+          memo = yield(memo, value) if index > 0
+        end
+        return memo
+      end
+    end
```

Comment:
**[REQUIRED]**
* Returns the same value as ruby's `inject` enumerable method. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from Odin).
* Combines all elements of enum by applying a binary operation, specified by a block.
![Screenshot from 2019-12-30 14-05-37](https://user-images.githubusercontent.com/44072711/71596536-b8031380-2b0d-11ea-9221-452683d4aee7.png)
* When a symbol is specified combines each element of the collection by applying the symbol as a named method.
![Screenshot from 2019-12-30 14-09-02](https://user-images.githubusercontent.com/44072711/71596615-ff899f80-2b0d-11ea-9677-1b24450b1ffc.png)